website/docs/fundamentals/exposing/ingress/external-dns.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "External DNS"
+sidebar_position: 30
+---
+
+External-DNS is a Kubernetes controller that automatically manages DNS records for your cluster's services and ingresses. It acts as a bridge between Kubernetes resources and DNS providers like AWS Route 53, 
+ensuring your DNS records stay synchronized with your cluster's state
+
+In this lab, we'll automate DNS management for Kubernetes Ingress resources using External-DNS with AWS Route 53
+
+We'll install External-DNS using Helm, the IAM Role ARN and Helm chart version provided as environment variables:
+
+```bash expectError=true
+helm repo add external-dns https://kubernetes-sigs.github.io/external-dns/
+helm upgrade --install external-dns external-dns/external-dns --version "${DNS_CHART_VERSION}" \
+    --namespace external-dns \
+    --create-namespace \
+    --set provider.name=aws \
+    --set serviceAccount.create=true \
+    --set serviceAccount.name=external-dns-sa \
+    --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"="$DNS_ROLE_ARN" \
+    --set txtOwnerId=eks-workshop \
+    --set extraArgs[0]=--aws-zone-type=private \
+    --set extraArgs[1]=--domain-filter=retailstore.com
+```
+
+Check that the External-DNS pod is running:
+
+```bash expectError=true
+$ kubectl -n external-dns get pods
+NAME                                READY   STATUS    RESTARTS   AGE
+external-dns-5bdb4478b-fl48s        1/1     Running   0          2m
+```
+
+Let's create an Ingress resource with DNS configuration:
+
+::yaml{file="manifests/modules/exposing/ingress/external-dns/ingress.yaml" paths="metadata.annotations.external-dns.alpha.kubernetes.io/hostname,spec.rules.0.host"}
+
+1. The annotation `external-dns.alpha.kubernetes.io/hostname` tells External-DNS which DNS name to create and manage for the Ingress, automating the mapping of your app’s hostname to its load balancer.
+2. The `spec.rules.host` defines the domain name your Ingress will listen to, which External-DNS uses to create a matching DNS record for the associated load balancer.
+
+Apply this configuration:
+
+```bash timeout=180 hook=add-ingress hookTimeout=430
```

Comment:
Don't think you meant to use this `hook`. You'll probably need a new one to block until the DNS is propagated otherwise the tests will fail.